File: python/190-ReverseBits/main.py
import unittest

class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0

        for i in range(32):
            result <<= 1
            result |= (n & 1)
            n >>= 1

        return result

class Tests(unittest.TestCase):
    def test_example1(self):
        solution = Solution()
        self.assertEqual(solution.reverseBits(0b00000010100101000001111010011100), 0b00111001011110000010100101000000)

if __name__ == '__main__':
    unittest.main()

File: cpp/190-ReverseBits/main.cpp
#include <iostream>

class Solution {
public:
    uint32_t reverseBits(uint32_t n) 
    {
        uint32_t result = 0;

        for(int i = 31; i >= 0 && n; i--)
        {
            result |= ((n & 1) << i);
            n >>= 1;
        }

        return result;    
    }
};

int main()
{
    Solution s;
    std::cout << s.reverseBits(0b00000010100101000001111010011100) << std::endl;
    std::cout << s.reverseBits(0b11111111111111111111111111111101) << std::endl;

    return 0;
}